### PR TITLE
Scripting

### PR DESCRIPTION
**Pull request by NicoRenaud**

### CODE REVIEW

File: swan/dataset/fingerprints_datasets.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    """Read the smiles, properties and compute the fingerprints."""
+    def __init__(self,
+                 data: str,
+                 properties: List[str] = None,
+                 type_fingerprint: str = 'atompair',
+                 fingerprint_size: int = 2048,
+                 sanitize=True) -> None:
+        """Generate a dataset using fingerprints as features.
+
+        Args:
```
Comment (felipeZ) [inline swan/dataset/fingerprints_datasets.py:23]: I use [Numpy Style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html) for the docs. We don't need to add the types in the documentation if we put them in the method (or function) definition

File: swan/dataset/graph_datasets.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    def _process(self):
+        pass
+
+    def __len__(self):
+        """Return dataset length."""
+        return len(self.molecules)
+
+    def __getitem__(self, idx):
+        """Return the idx dataset element."""
+        return self.norm(self.molecular_graphs[idx])
```
Comment (felipeZ) [inline swan/dataset/graph_datasets.py:95]: Do you want to apply the normalization always? I don't think it is a problem if the data is already normal and we apply a second normalization

File: swan/dataset/fingerprints_datasets.py (hunk 1 above)
Comment (felipeZ) [reply swan/dataset/fingerprints_datasets.py:23]: I can change them :)

File: swan/dataset/graph_datasets.py (hunk 2 above)
Comment (NicoRenaud) [reply swan/dataset/graph_datasets.py:95]: this normalize is kinda weird cause it 'row normalize the feature so that they equal to 1' But dependind on the distribution of the invidual feartures we might change them in almost random fashion .... Or maybe I don't understand correctly

Comment (NicoRenaud) [reply swan/dataset/graph_datasets.py:95]: But I would say we may not want to use norm

File: swan/modeller/modeller_base.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+        # current number of epoch
+        self.epoch = 0
+
+    def set_optimizer(self, name, *args, **kwargs):
+        """Set an optimizer using the config file
+
+        Args:
+            name ([type]): [description]
+        """
+        self.optimizer = torch.optim.__getattribute__(name)(
```
Comment (felipeZ) [inline swan/modeller/modeller_base.py:84]: I am not a big fun of magic methods, mostly because I forget all the time how they interact with the other methods.
 Is this `__getattribute__` call equivalent to: `getattr(torch.optim,name)`?

File: swan/modeller/modeller_base.py [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+from abc import abstractmethod
+
+from pathlib import Path
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+import torch
+
+from flamingo.utils import Options
```
Comment (felipeZ) [inline swan/modeller/modeller_base.py:15]: This import is not used

File: swan/modeller/modeller_base.py [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+from sklearn.preprocessing import RobustScaler
+from torch import Tensor, nn
+from torch.utils.data import DataLoader, Dataset
+
+from ..utils.early_stopping import EarlyStopping
+
+from ..utils.input_validation import MINIMAL_MODELER_DEFAULTS
```
Comment (felipeZ) [inline swan/modeller/modeller_base.py:23]: These defaults are never used

File: swan/modeller/modeller_base.py [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+    def validate_model(self) -> Tuple[Tensor, Tensor]:
+        """Evaluate the model against the validation dataset."""
+        results = []
+        expected = []
+
+        # Disable any gradient calculation
+        with torch.no_grad():
+            self.network.eval()
+            loss_all = 0
+            for x_val, y_val in self.valid_loader:
```
Comment (felipeZ) [inline swan/modeller/modeller_base.py:160]: the `self.valid_loader` is not defined in the base class. Could you please initialize it to `None`.

File: swan/modeller/modeller_base.py [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+        self.set_optimizer('SGD', lr=0.01)
+
+        # set the default loss
+        self.set_loss('MSELoss')
+
+        # set scheduler
+        self.set_scheduler('StepLR', 0.1)
+
+        # I/O options
+        self.workdir = './'
```
Comment (felipeZ) [inline swan/modeller/modeller_base.py:72]: Can we define the `workdir` as:
```py
self.workdir = Path(".")
```
Like a Path for the whole class.

File: swan/modeller/fingerprints_modeller.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import numpy as np
```
Comment (felipeZ) [inline swan/modeller/fingerprints_modeller.py:1]: This import is never used

File: swan/modeller/fingerprints_modeller.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+from torch.utils.data import DataLoader, random_split
+from .modeller_base import ModellerBase
+
+
+class FingerprintModeller(ModellerBase):
+    """Object to create models using fingerprints."""
+    def create_data_loader(self,
+                           frac=[0.8, 0.2],
+                           batch_size: int = 64) -> DataLoader:
```
Comment (felipeZ) [inline swan/modeller/fingerprints_modeller.py:10]: The returning type of this method is `None`

File: swan/modeller/graph_modeller.py [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+import numpy as np
+
+import torch
+from torch import Tensor
+import torch_geometric as tg
+
+from torch.utils.data import DataLoader, random_split
+
+from ..dataset import MolGraphDataset
+from .modeller_base import ModellerBase
```
Comment (felipeZ) [inline swan/modeller/graph_modeller.py:14]: There are a bunch of import statements that are not used

File: swan/modeller/modeller_base.py (hunk 4 above)
Comment (felipeZ) [reply swan/modeller/modeller_base.py]: removed

File: swan/dataset/fingerprints_datasets.py (hunk 1 above)
Comment (felipeZ) [reply swan/dataset/fingerprints_datasets.py]: I have changed them to Numpy style

File: swan/dataset/graph_datasets.py (hunk 2 above)
Comment (NicoRenaud) [reply swan/dataset/graph_datasets.py]: I made that optional so we can deal with it later

**Review by felipeZ — APPROVED** [review]
Great Job!
